### PR TITLE
[WIP] Support Clementine Remote Protocol

### LINKED ISSUE #454: Support Clementine remote control protocol
This would allow users to remote control the player from an Android device, for instance.

See [`clementine-remote`](https://github.com/emersion/node-clementine-remote).

### PR DESCRIPTION
**Pull request by emersion**
Fixes #454

### CODE REVIEW

File: lib/clementine_server.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+  var player = this.player;
+
+  var server = new ClementineServer({
+    host: host,
+    port: port
+  });
+
+  // Current position
+  // TODO: something more reliable
+  var position = 0, positionInterval;
```
Comment (andrewrk) [inline lib/clementine_server.js]: How does the position work? In Groove Basin protocol, the server sends the current server time and what time the song was started, or what time the song was paused.

**Comment by andrewrk** [comment]
Cool, this is looking easy enough to support.

File: lib/clementine_server.js (hunk 1 above)
Comment (emersion) [inline lib/clementine_server.js]: The way it works in Clementine is

Comment (andrewrk) [inline lib/clementine_server.js]: hmm this is not very well designed. I kind of want to suggest that we go bug clementine to support the groove basin protocol instead :-/

Comment (diogocp) [inline lib/clementine_server.js]: "not very well designed" is a bit of an understatement. I have noticed that Clementine Remote drains my battery very fast while connected to the server. This explains a lot.

**Pull request closed without merging**